Gateway/Command/AbstractCommand.php

```diff
@@ -0,0 +1,342 @@
+            return;
+        }
+
+        $this->logger->error(
+            sprintf(
+                "Failed to %s order %s with SwedbankPay payment id %s, response resource:\n%s",
+                $command,
+                $mageOrder->getEntityId(),
+                $paymentId,
+                // phpcs:ignore Magento2.Functions.DiscouragedFunction
+                print_r($responseResource, true)
+            )
+        );
+
+        throw new SwedbankPayException(
+            new Phrase(
+                sprintf(
+                    "SwedbankPay %s Error: Failed to parse response for SwedbankPay payment %s.",
+                    ucfirst($command),
+                    $paymentId
+                )
+            )
+        );
+    }
+
+    /**
+     * @param string $command
+     * @param array $responseData
+     * @param OrderInterface $mageOrder
+     * @param string $paymentId
+     * @throws SwedbankPayException
+     */
+    protected function checkResponseData($command, $responseData, $mageOrder, $paymentId)
+    {
+        if (is_array($responseData)
+            && isset($responseData[$this->cmdTransActionMap[$command]]['transaction']['state'])
+            && $responseData[$this->cmdTransActionMap[$command]]['transaction']['state'] == 'Completed') {
+            return;
+        }
+
+        $this->logger->error(
+            sprintf(
+                "Failed to %s order %s with SwedbankPay payment order id %s, response data:\n%s",
+                $command,
+                $mageOrder->getEntityId(),
+                $paymentId,
+                // phpcs:ignore Magento2.Functions.DiscouragedFunction
+                print_r($responseData, true)
+            )
+        );
+
+        throw new SwedbankPayException(
+            new Phrase(
+                sprintf(
+                    "SwedbankPay %s Error: Failed to %s SwedbankPay payment order %s.",
+                    ucfirst($command),
+                    $command,
+                    $paymentId
+                )
+            )
+        );
+    }
+
+    /**
+     * Get SwedbankPay payment order data
+     *
+     * @param MagentoOrderInterface|int|string $order
+     *
+     * @return PaymentOrderInterface|PaymentQuoteInterface|false
+     * @throws NoSuchEntityException
+     */
+    protected function getSwedbankPayPaymentData($order)
+    {
+        if (is_numeric($order)) {
+            return $this->paymentOrderRepo->getByOrderId($order);
+        }
+
+        if ($order instanceof MagentoOrderInterface) {
+            if ($order->getEntityId()) {
+                return $this->paymentOrderRepo->getByOrderId($order->getEntityId());
+            }
+            return $this->paymentQuoteRepo->getByQuoteId($order->getQuoteId());
+        }
+
+        $this->logger->error(
+            // phpcs:ignore Magento2.Functions.DiscouragedFunction
+            sprintf("Unable to find a SwedbankPay payment matching order:\n%s", print_r($order, true))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The use of function print_r() is discouraged](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-magento2-payments/pullRequest?prid=5056328)